### ISSUE TITLE
OneClassSVM predicts wrong number of classes + wrong predictions

### ISSUE THREAD

**Issue by naf3a** [onnx/sklearn-onnx#300]
According to the [scikit-learn documentation](https://scikit-learn.org/stable/modules/generated/sklearn.svm.OneClassSVM.html#sklearn.svm.OneClassSVM.predict), the `predict` method of `OneClassSVM`  should return  **+1** or **-1**.

However, when I run this model with `onnxruntime`, **it returns 0 also !**

Specially when the specified kernel is `sigmoid` and sometimes when the kernel is `linear` or `poly`.

On the other hand, saved model doesn't give the same results as the predict method. 
Example:

```python
from skl2onnx.common.data_types import Int64TensorType
from skl2onnx import convert_sklearn
from sklearn.svm import OneClassSVM
import onnxruntime as rt
import numpy as np

X = np.array([[0, 1, 2], [44, 36, 18], [-45, -78, -46]])

linearOCSVM = OneClassSVM(kernel='sigmoid').fit(X)
print(linearOCSVM.predict(X)) # [ 1  1 -1]

# Convert into ONNX format with onnxmltools
initial_type = [('int64_input', Int64TensorType([None, X.shape[1]]))]
onx = convert_sklearn(linearOCSVM, initial_types=initial_type)

# Export model into ONNX format
with open("linearOCSVM.onnx", "wb") as f:
    f.write(onx.SerializeToString())

# Compute the prediction with ONNX Runtime
sess = rt.InferenceSession("linearOCSVM.onnx")
input_name = sess.get_inputs()[0].name
label_name = sess.get_outputs()[0].name
pred_onx = sess.run([label_name], {input_name: X})[0]
print(pred_onx) # [[ -1] [ 1 ] [ 1 ]]
```

> Not a native speaker, please forgive me if I made english mistakes.

**Comment by prabhat00155** [comment]
@xadupre Could you take a look into this?

**Comment by xadupre** [comment]
I looked into it and the values -1, 1 are the sign of the scores and these scores (decision_function) are very close to zero in this particular example. The errors are due to float32 approximation of float64.

**Comment by xadupre** [comment]
I confirmed by running a version of the runtime of SVMRegressor implemented in onnxruntime tweaked for doubles and the predictions exactly match scikit-learn's ones.
